.github/workflows/integration-tests.yml

```diff
@@ -36,24 +52,29 @@ jobs:
         -f pull_request_number=${{ github.event.pull_request.number }} \
         -f commit_sha=${{ github.event.pull_request.head.sha }} 
 
-    
-
   # Statuses and checks apply to specific commits (by hash). 
   # Enforcement of required checks is done both at the PR level and the merge queue level.
   # In case of multiple commits in a single PR, the hash of the squashed commit 
   # will not match the one for the latest (approved) commit in the PR.
   # We auto approve the check for the merge queue for two reasons:
   # * Queue times out due to duration of tests.
   # * Avoid running integration tests twice, since it was already run at the tip of the branch before squashing.
+  # Additionally, integration tests are not run on docs-only changes.
   auto-approve:
-    if: github.event_name == 'merge_group'
+    needs: check-integration-test-changes
+    if: '{{ github.event_name == "merge_group" || env.integration_tests_required == "false" }}'
```

Comment:
Actually, this won't work in the merge queue. If the "need" job is skipped, this will be skipped too (even with the "if" condition). You can use always() to ensure it runs. Also, use the output
```if: always() && (github.event_name == 'merge_group' || needs.check-integration-test-changes.outputs.integration_tests_required == 'true')```